x-pack/plugins/viz_editor/public/pseudo_plugins/xy_chart_plugin/xaxis_editor.tsx

```diff
@@ -19,21 +18,26 @@ export function XAxisEditor({
   visModel: any;
   onChangeVisModel: (visModel: VisModel) => void;
 }) {
-  const onDropField = (field: DatasourceField) => {
-    const operation: SelectOperation =
-      field.type === 'date'
-        ? { operator: 'date_histogram', argument: { field: field.name, interval: '1m' } }
-        : { operator: 'column', argument: { field: field.name } };
+  const column = selectColumn(col, visModel);
 
-    onChangeVisModel(updateColumn(col, operation, visModel));
-  };
+  if (!column) {
+    // TODO...
+    return <span>N/A</span>;
+  }
 
   return (
-    <Draggable
-      canHandleDrop={(f: DatasourceField) => f && (f.type === 'string' || f.type === 'date')}
-      onDrop={onDropField}
+    <OperationEditor
+      column={column}
+      visModel={visModel}
+      onColumnChange={newColumn => {
+        onChangeVisModel(updateColumn(col, newColumn, visModel));
+      }}
+      allowedScale="ordinal"
```

Comment:
Linear and date scales should be supported on the X axis too... this appears buggy when I checked it out
<img width="208" alt="Screenshot 2019-04-09 18 48 50" src="https://user-images.githubusercontent.com/666475/55840071-1f98af00-5af8-11e9-9ec6-f30b6413110a.png">

Reply:
Interval/ratio values are also ordinal, so this is inclusive. How did you get to the screenshot in your comment? This works fine for me:
![xaxis_switch](https://user-images.githubusercontent.com/1508364/55869653-39350780-5b87-11e9-8329-00d54e4caafe.gif)

Reply:
I got there by building it from a suggestion

Reply:
Then it's a bug in the `getSuggestion` function, not the axis editing itself. There are some instabilities in there, but we should address that in a separate PR.